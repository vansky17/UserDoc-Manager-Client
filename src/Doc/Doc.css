tr.tr-toggable {
    box-shadow: inset 0 3px 3px -2px rgba(0, 0, 0, .7);
    /* background-color: #dce4de; */
    /* box-shadow: inset 0 0 8px #000000; */
     cursor:pointer;
    
}
tr.tr-toggable td{
    padding: 0.3em 0;
}
 tr.expandable > td {
  box-shadow: inset 0 6px 7px -4px rgba(0, 0, 0, .7);
  padding: 0;
}

tr.expandable > td > .inner {
  margin: 5px;
  overflow: hidden;
} 
tbody {
  box-shadow: 1px 1px 7px 1px rgba(0,0,0,0.6);
}