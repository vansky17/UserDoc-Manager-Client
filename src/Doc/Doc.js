import React from 'react'
import { slideDown, slideUp } from '../anim';
import { format } from 'date-fns'
import './Doc.css'


class Doc extends React.Component {
 
  state = { expanded: false }

  toggleExpander = (e) => {
    if (e.target.type === 'checkbox') return;

    if (!this.state.expanded) {
      this.setState(
        { expanded: true },
        () => {
          if (this.refs.expanderBody) {
            slideDown(this.refs.expanderBody);
          }
        }
      );
    } else {
      slideUp(this.refs.expanderBody, {
        onComplete: () => { this.setState({ expanded: false }); }
      });
    }
  }
  
 render () {
  const { name, partnum,vernum, formattype, reldate, author, descr, path} = this.props 

  return [
    <tr onClick={this.toggleExpander} className="tr-toggable">
 <td><input style={{display: "none"}} type="checkbox" />{name}</td>
  <td >{partnum}</td>
  <td>{vernum}</td>
  <td>{formattype}</td>
  <td>{format(reldate, 'Do MMM YYYY')}</td>
    </tr>,
    this.state.expanded && (
      <tr className="expandable" key="tr-expander">
        <td className="description-background" colSpan={7}>
          <div ref="expanderBody" className="inner uk-grid">
            <div >
              <h4>Description</h4>
              <ul> 
                <li className="descr-text">{descr}</li>
                <li>Author: {author} </li>
                <li><button className="open-doc"><a href={path} target="_blank" rel="noopener noreferrer">OPEN</a></button></li>
              </ul>
            </div>
          </div>
        </td>
      </tr>
      )
    ];
  }
}

export default Doc