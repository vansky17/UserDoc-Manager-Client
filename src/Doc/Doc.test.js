import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Doc from './Doc'


describe(`Doc component`, () => {
  const props = {
     name: "name",
     partnum: "partnum",
     vernum: 1, 
    formattype: "PDF",
    reldate: new Date(2018, 12, 15), 
    author: 'IS',
    descr: "No description",
    path: "google.com" 
}

  it('renders a .Doc by default', () => {
    const wrapper = shallow(<Doc />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Doc given props', () => {
    const wrapper = shallow(<Doc {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})