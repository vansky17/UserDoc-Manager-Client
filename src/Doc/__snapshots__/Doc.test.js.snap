// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doc component renders a .Doc by default 1`] = `
Array [
  <tr
    className="tr-toggable"
    onClick={[Function]}
  >
    <td>
      <input
        style={
          Object {
            "display": "none",
          }
        }
        type="checkbox"
      />
    </td>
    <td />
    <td />
    <td />
    <td>
      Invalid Date
    </td>
  </tr>,
  "",
]
`;

exports[`Doc component renders the Doc given props 1`] = `
Array [
  <tr
    className="tr-toggable"
    onClick={[Function]}
  >
    <td>
      <input
        style={
          Object {
            "display": "none",
          }
        }
        type="checkbox"
      />
      name
    </td>
    <td>
      partnum
    </td>
    <td>
      1
    </td>
    <td>
      PDF
    </td>
    <td>
      15th Jan 2019
    </td>
  </tr>,
  "",
]
`;
